jit.js

```diff
@@ -692,7 +932,12 @@ to single-step.
     generateLabel: function() {
         // remember label position for deleteUnneededLabels()
         this.sourceLabels[this.prevPC] = this.source.length;
-        this.source.push("case ", this.prevPC, ":\n");
+        if (!this.suppressNextLabel) {
+            this.source.push("case ", this.prevPC, ":\n");
+        } else {
+            this.source.push("\n");
+        }
+        this.suppressNextLabel = false;
```

Comment:
Why this extra-mechanism of suppressing labels? Why not remove the label at the end, as with all other unneeded labels?